### PR TITLE
Add form step completed event

### LINKED ISSUE #715: submitted Event updates in PaymentProvisioning component
We will likely change the nature of the `submitted` event emission for this component in our next major release. See - [insert issue link here]

At the moment though we want to make usage of this event more clear without implementing breaking changes. 

Actions:

- Update `submitted` event emission on each form step to include the name of the completed step in a metaData field or something similarly named
- Update documentation for `submitted` on this component to show and explain how the metaData property can be used to identify what request the submitted event was emitting.

### PR DESCRIPTION
**Pull request by colganmeanor**
### Add new event to payment-provisioning: `form-step-completed`

In our 5.x.x release, we plan to adjust the expected behavior and payload of our `submitted` event in the `payment-provisioning` component. We plan to update the behavior of `submitted` so that it is only emitted at the very end of the component form UI flow. This will bring its behavior in line with our other components that also emit a `submitted` event. 

This change will be considered a breaking change, which is why we are waiting until our 5.x.x release to commit this change. In the mean time, this PR begins preparing for this change by adding a new event to the `payment-provisioning` component: `form-step-completed`

This new event will be emitted alongside `submitted` for now, until we alter the `submitted` event's behavior to only emit at the very end of the form flow. 

The story documentation and example have been updated as well to reflect the new event. It is also captured on the storybook action panel to view in storybook when it is emitted. 

Links
-----

Closes #715 

Developer considerations
--------------

- On any task
  - [ ] Previous unit and e2e tests are passing
  - [ ] Perform dev QA on Chrome, Firefox, and Safari
- On new features
  - [ ] Add unit tests
  - [ ] Add e2e tests
  - [ ] Add documentation
  - [ ] Does the component have exportedparts for styling? If so, add unit a unit test for each exportedpart
- On bugs
  - [ ] Add unit tests to prevent the bug from happening again

Developer QA steps
--------------------

- [ ] Each step of the payment provisioning component should now emit `form-step-completed` event. 
  - [ ] `Event.detail.data` contains the server response from the form step
  - [ ] `Event.detail.formStep` contains the name of the completed form step
- [ ] New event documentation in storybook looks OK
- [ ] Example usage in storybook docs looks OK

### CODE REVIEW

File: apps/docs/stories/components/PaymentProvisioning/index.stories.tsx [hunk 1]
```diff
@@ -57,6 +57,16 @@ const meta: Meta = {
         category: "events",
       },
     },
+    'form-step-completed': {
+      description: "Emitted when a form step is completed. Contains data from that form step's server response. The name of the completed form step is defined in `data.detail.formStep`.",
```
Comment (Claudia108) [inline apps/docs/stories/components/PaymentProvisioning/index.stories.tsx:61]: Could you add this to the first sentence: `... after the user clicks 'Next'.`

**Review by Claudia108 — COMMENTED** [review]
Awesome! You are a machine. Just added a tiny comment.

**Comment by TiagoFuelber** [comment]
On Storybook actions addon, I can see the event being emitted in all steps, except from 6 to 7 and from 7 to 8.

**Review by TiagoFuelber — APPROVED** [review]
Awesome! Approved

**Pull request merged**